Port MASTG-TEST-0066 (iOS TLS Settings) to MASTG v2 (#3792)
OWASP/mastg: tests-beta/ios/MASVS-NETWORK/MASTG-TEST-0x03.md

Since ATS does not apply to Network.framework connections, any weak TLS configuration here is not mitigated by ATS. Setting a minimum TLS version below 1.2 creates the risk of connection downgrade attacks without any ATS-level safety net. Apple's documentation notes that "ATS doesn't apply to calls your app makes to lower-level networking interfaces like the `Network` framework or `CFNetwork`. In these cases, you take responsibility for ensuring the security of the connection." See [Preventing Insecure Network Connections](https://developer.apple.com/documentation/security/preventing-insecure-network-connections).

For more information on iOS network APIs and when ATS applies, see @MASTG-KNOW-0071 and @MASTG-KNOW-0073.

## Steps

1. Use @MASTG-TECH-0065 to reverse engineer the app.

> 2. Use @MASTG-TECH-0072 to look for uses of `sec_protocol_options_set_min_tls_protocol_version` and `sec_protocol_options_set_max_tls_protocol_version` in the app binary.

Reviewer: ```suggestion
2. Use @MASTG-TECH-0066 to look for uses of `sec_protocol_options_set_min_tls_protocol_version` and `sec_protocol_options_set_max_tls_protocol_version` in the app binary.
```

3. Use @MASTG-TECH-0076 to analyze the relevant code paths and determine the TLS version values passed to those functions.

## Observation

The output should contain any calls to TLS protocol version configuration functions in the Network.framework, if found.

## Evaluation

The test case fails if the app calls:

- `sec_protocol_options_set_min_tls_protocol_version` with a value of `tls_protocol_version_TLSv10` (`0x0301`) or `tls_protocol_version_TLSv11` (`0x0302`), or
- `sec_protocol_options_set_max_tls_protocol_version` with a value of `tls_protocol_version_TLSv10` (`0x0301`) or `tls_protocol_version_TLSv11` (`0x0302`).

> Because Network.framework operates entirely outside of ATS, a connection configured this way will succeed against a server that supports the deprecated TLS version, with no ATS-level safety net.

Reviewer: ```suggestion
Because Network.framework operates entirely outside of ATS, a connection configured this way will succeed against a server that supports the deprecated TLS version, bypassing ATS.
```
